### DISCUSSION TITLE
volume.vacuum and erasure coding shards

### DISCUSSION THREAD

**Discussion by steven2612** [seaweedfs/seaweedfs#5168] [Ideas]
Hi,

i know compacting on ec shards is currently not implemented.
so with volume.vacuum those shards are simply ignored.

you can manually do a ec.decode -volumeId XX and then a volume.vacuum

wouldn't this be an option to do with volume.vacuum automatically?
so:
 - check if threshold is reached
 - do ec.decode
 - do compacting 

volume.list also seems not to display the currently deleted bytes of an ec shard - is it possible to see this anywhere ?

best regards
sg

**Comment by gada121982** [comment]
That is good idea. I think we should add a option to decode the ec volume that has too much deleted files only. Because ec.decode/ec.encode may reduce system performance. @steven2612
